ra/ra.go

```diff
@@ -72,6 +74,93 @@ type RegistrationAuthorityImpl struct {
 	totalCertsStats      metrics.Scope
 }
 
+// Note: the issuanceExpvar must be a global. If it is a member of the RA, or
+// initialized with everything else in NewRegistrationAuthority() then multiple
+// invocations of the constructor (e.g from unit tests) will panic with a "Reuse
+// of exported var name:" error from the expvar package.
+var issuanceExpvar = expvar.NewInt("successfulIssuances")
+
+// Wrap time.Tick so we can override it in tests.
+var makeTicker = func(tickerDuration time.Duration) (func(), <-chan time.Time) {
+	t := time.NewTicker(tickerDuration)
+	return t.Stop, t.C
+}
+
+// The issuanceLoadStat allows us to expose a window of RA issuance volume as an
+// expvar. Internally a circular slice of 60 counts is maintained (one for each
+// minute). When a certificate is issued we increment the count in the current
+// minutes' bucket. A Go routine monitoring a ticket updates the expvar
+// periodically with the past 5 minutes of buckets totalled up.
+type issuanceLoadStat struct {
+	sync.RWMutex
+	issuedCounts []int
+	windowSize   int
+	ticker       <-chan time.Time
+	clk          clock.Clock
+}
+
+// Spawn a Go routine blocking on the ticker
+func (i *issuanceLoadStat) updateForever() {
+	go func() {
+		for {
+			// Compute the count and update the expvar immediately so the
+			// first round of the go-routine sets the initial expvar value
+			count := i.countIssuances()
+			issuanceExpvar.Set(int64(count))
+			// Then block on the ticker until its time to update again
+			<-i.ticker
+		}
+	}()
+}
+
+func (i *issuanceLoadStat) countIssuances() int {
+	// Calculate a sum over the last `windowSize` buckets of the `issuedCounts`
+	// slice, wrapping around as required to stay within the 60 buckets.
+	sum := 0
+	cur := i.clk.Now().Minute()
+	i.RLock()
+	for j := i.windowSize - 1; j >= 0; j-- {
+		index := (cur - j) % 60
+		// Note: Go's % operator is remainder, not modulus and we need to handle the
+		// negative case ourselves
+		if index < 0 {
+			index = index + 60
+		}
+		sum += i.issuedCounts[index]
+	}
+	i.RUnlock()
```

Comment:
Typically we put `defer i.RUnlock()` right after the lock is acquired so it's easy to see both are there.

Also, let's call the function receiver `ils` instead of `i` to avoid confusion with an index variable.